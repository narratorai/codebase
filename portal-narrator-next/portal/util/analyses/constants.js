export const ICON_BOX_WIDTH = '120px'
export const CONTENT_MAXWIDTH = '832px'
export const INSIGHT_MAXWIDTH = '920px'
export const COPY_MAXWIDTH = '750px'
export const METRIC_MAXWIDTH = '960px'

export const STREAM_UPDATE_KIND = 'on_stream_update'
export const CRON_UPDATE_KIND = 'cron'
