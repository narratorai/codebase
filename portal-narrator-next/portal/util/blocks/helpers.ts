import _ from 'lodash'
import { ObjectFieldTemplateProps } from '@rjsf/core'
import {
  BlockContent,
  BlockContentType,
  IFormContext,
  RefreshConfig,
  GroupedTab,
  TabConfig,
  TabsConfig,
  FieldConfig,
} from './interfaces'

export const triggerSchemaAndDataUpdates = (formContext: IFormContext, options: RefreshConfig, id: string) => {
  if (
    (options.update_schema || options.process_data || options.submit_form) &&
    formContext?.onRefreshConfigRequest &&
    formContext?.version === 1
  ) {
    formContext.onRefreshConfigRequest({
      field_slug: id,
      ...options,
    })
  }
}

export const findBlockContentByType = <T extends BlockContent>(
  type: BlockContentType,
  blockContent: BlockContent[] | null
): T | undefined => {
  return blockContent?.find((block) => block.type === type) as T
}

export const findAllBlockContentByType = <T extends BlockContent[]>(
  type: BlockContentType,
  blockContent: BlockContent[] | null
): T | undefined => {
  return _.compact(
    _.map(blockContent, (block) => {
      if (block.type === type) return block
      return null
    })
  ) as T
}

// Tab support in ObjectFieldTemplate
export const groupPropertiesByTab = (
  tabsConfig: TabsConfig,
  properties: ObjectFieldTemplateProps['properties']
): GroupedTab[] => {
  // Use the property_names from each tabConfig to assemble root properties in each tab:
  const selectedTabs = _.reduce(
    tabsConfig.tabs,
    (result: any, tab: TabConfig) => {
      const selectedProperties = _.filter(properties, (property) => _.includes(tab.property_names, property.name))
      return [
        ...result,
        {
          ...tab,
          label: tab.label,
          properties: selectedProperties,
          tab_id: tab.tab_id,
          redirect_tab_ids: tab.redirect_tab_ids,
        },
      ]
    },
    []
  )

  return selectedTabs
}

export const getDependentFieldsForField = ({
  fieldConfig,
  fieldConfigs,
}: {
  fieldConfig: FieldConfig
  fieldConfigs: FieldConfig[]
}): string[] => {
  // `map` through all fields to get all dependencies of a given field
  return _.compact(
    _.map(fieldConfigs, (f) => {
      if (_.includes(f.field_depends_on, fieldConfig.name)) {
        return f.name
      }
    })
  )
}

export const getAllDependentFields = (fields: FieldConfig[]) => {
  // - `filter` all fields that have `field_depends_on`
  // - `map` through the fields and get all of the `field_depends_on` values
  // - `flatten` the nested arrays (since `field_depends_on` is an array of values)
  // - `uniq` the end result to rid of duplicates
  const fieldDependsOn = fields.filter((f) => f.field_depends_on).map((field) => field.field_depends_on || [])
  return _.uniq(_.flatten(fieldDependsOn))
}
