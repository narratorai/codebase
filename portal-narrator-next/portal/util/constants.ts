// Breakpoints https://github.com/jxnblk/grid-styled#breakpoints
export const breakpoints = {
  sm: '32em', // (512px)
  md: '48em', // (768px)
  tablet: '64em', // (1024px)
  lg: '75em', // (1200px)
}

export const zIndex = {
  select: 999999,
  closeChatWidget: 999998,
  overlay: 1000,
  notification: 1000002,
}

type Color = {
  [key: string]: string
}

// https://projects.invisionapp.com/share/8WEP8J5SM#/screens/270372538
export const colors: Color = {
  white: '#ffffff',
  black: '#242424',

  // gray
  gray100: '#f9f9f9',
  gray200: '#f4f4f4',
  gray300: '#e5e5e5',
  gray400: '#cfd0d0',
  gray500: '#979899',
  gray600: '#666768',
  gray700: '#505152',
  gray800: '#333435',
  gray900: '#1d1d1e',

  // blue
  blue100: '#f1f7fb',
  blue200: '#dcedf8',
  blue300: '#99d0f3',
  blue400: '#58b3eb',
  blue500: '#298dcc',
  blue600: '#1d6c9e',
  blue700: '#0e4b71',
  blue800: '#001e3e',
  blue900: '#001021',

  // purple
  purple100: '#f6f1fe',
  purple200: '#e2d7f2',
  purple300: '#d0b7f5',
  purple400: '#a980e4',
  purple500: '#764cb2',
  purple600: '#513a73',
  purple700: '#382554',
  purple800: '#240f41',
  purple900: '#14022c',

  // red
  red100: '#feeeee',
  red200: '#f8c8c8',
  red300: '#ff9f9f',
  red400: '#f77374',
  red500: '#f04e4f',
  red600: '#be2728',
  red700: '#850d0e',
  red800: '#490101',
  red900: '#260000',

  // magenta
  magenta100: '#f9f2f6',
  magenta200: '#f6d6e8',
  magenta300: '#f3a6d2',
  magenta400: '#e480b9',
  magenta500: '#cd3c8e',
  magenta600: '#a71f6c',
  magenta700: '#790848',
  magenta800: '#51002e',
  magenta900: '#270016',

  // yellow
  yellow100: '#fdf6e5',
  yellow200: '#fcedc6',
  yellow300: '#ffdd8b',
  yellow400: '#ffd160',
  yellow500: '#ffb025',
  yellow600: '#d38c26',
  yellow700: '#a0550c',
  yellow800: '#472404',
  yellow900: '#221100',

  // orange
  orange100: '#fff4ec',
  orange200: '#f6d9c4',
  orange300: '#ffbd8c',
  orange400: '#ffa461',
  orange500: '#ff8125',
  orange600: '#ae5310',
  orange700: '#6f2f00',
  orange800: '#431f04',
  orange900: '#230f00',

  // green
  green100: '#ecfef4',
  green200: '#ccf2dc',
  green300: '#91f0b9',
  green400: '#52e18e',
  green500: '#20b05c',
  green600: '#117c3e',
  green700: '#065326',
  green800: '#003315',
  green900: '#001b0b',

  // teal
  teal100: '#eefbfa',
  teal200: '#d0f1ed',
  teal300: '#97eee2',
  teal400: '#50e3cf',
  teal500: '#1da290',
  teal600: '#0f7769',
  teal700: '#05584d',
  teal800: '#002d27',
  teal900: '#001815',

  // lime
  lime100: '#f6feef',
  lime200: '#bceb90',
  lime300: '#a4d972',
  lime400: '#8ac751',
  lime500: '#6bb328',
  lime600: '#437417',
  lime700: '#2f550d',
  lime800: '#1c3406',
  lime900: '#0f1e01',

  // blurple
  blurple100: '#f3f3ff',
  blurple200: '#c0bbfc',
  blurple300: '#9b94f3',
  blurple400: '#6f65f6',
  blurple500: '#5a4ef0',
  blurple600: '#2c2589',
  blurple700: '#201a67',
  blurple800: '#0d0a35',
  blurple900: '#080624',

  // V2 MAVIS AI COlORS
  mavis_black: '#0A0519',
  mavis_light_gray: '#F2F1F4',
  mavis_text_gray: '#9B9A9E',
  mavis_dark_gray: '#5A595C',
  mavis_darker_gray: '#E7E3EE',
  mavis_off_white: '#F7F7F7',
  mavis_checkout_blue: '#1586E8',
  mavis_subscribed_green: '#20CB49',
  mavis_light_purple: '#D759F7',
  mavis_dark_purple: '#6331B3',
}

export const PINK_HEART_COLOR = '#eb2f96'
export const PERCENT_OF_TOTAL_COLOR = '#353800'

export const NOT_IN_COHORT_COLOR = colors.blue500

// FONT INDEXES       //  0   1   2   3   4   5   6   7   8   9
export const fontSizes = [10, 11, 13, 14, 15, 18, 22, 26, 30, 42]

// FIXME - These lineHeights are deprecated, please use typography
export const lineHeights = [
  '14px', // 0 -- with 11px fontSize
  '16px', // 1 -- with 13px fontSize
  '20px', // 2 -- with 16px fontSize
  '24px', // 3 -- with 20px fontSize
  '30px', // 4 -- with 26px fontSize
  '46px', // 5 -- with 36px fontSize
]

export const lightWeight = 300
export const semiBoldWeight = 600

export const rebassTheme = {
  breakpoints: [breakpoints.sm, breakpoints.md, breakpoints.tablet, breakpoints.lg],
  colors,
  fonts: {
    sans: '"Source Sans Pro", sans-serif',
  },
  fontSizes,
  fontWeights: {
    normal: 400,
    semiBold: semiBoldWeight,
    bold: 700,
  },
  space: [0, 8, 16, 24, 32, 40, 48, 56, 64],
  zIndex,
}

export const MAX_WIDTHS = [360, 768, 1336, 1336, 1920]
export const MAX_WIDTH = 1366
export const PLOT_HEIGHT = 480
export const SIDENAV_WIDTH = 200
export const SIDENAV_WIDTH_COLLAPSED = 66
export const SIDEBAR_WIDTH = 240
export const MOBILE_TOPNAV_HEIGHT = 56
export const LAYOUT_SIDER_WIDTH = 360

export const STATUS_PAGE_BANNER_Z_INDEX = 99
export const DEFAULT_LOGIN_REDIRECT_PATH = '/datasets'
export const DEFAULT_LOGIN_REDIRECT_PATH_V2 = '/chat'
export const DEFAULT_MOBILE_LOGIN_REDIRECT_PATH = '/narratives'

export const typography = {
  // MARKDOWN
  // "heading" values taken from Chrome initial values
  heading1: {
    sizes: ['2em'],
    lineHeights: ['initial'],
    fontWeight: 'semiBold',
  },
  heading2: {
    sizes: ['1.5em'],
    lineHeights: ['initial'],
    fontWeight: 'semiBold',
  },
  heading3: {
    sizes: ['1.17em'],
    lineHeights: ['initial'],
    fontWeight: 'semiBold',
  },
  heading4: {
    sizes: ['initial'],
    lineHeights: ['initial'],
    fontWeight: 'semiBold',
  },

  // BODY
  body400: {
    sizes: [fontSizes[0]], // 10
    lineHeights: [1.2],
  },
  body300: {
    sizes: [fontSizes[1]], // 11
    lineHeights: [1.28],
  },
  body200: {
    sizes: [fontSizes[2]], // 13
    lineHeights: [1.24],
  },
  body100: {
    sizes: [fontSizes[3]], // 14
    lineHeights: [1.43],
  },
  body50: {
    sizes: [fontSizes[4]], // 15
    lineHeights: [1.52],
  },

  // TITLES
  title400: {
    sizes: [fontSizes[5]], // 18
    lineHeights: [1.34],
  },
  title300: {
    sizes: [fontSizes[6]], // 22
    lineHeights: [1.28],
  },
  title200: {
    sizes: [fontSizes[7]], // 26
    lineHeights: [1.24],
  },
  title100: {
    sizes: [fontSizes[8]], // 30
    lineHeights: [1.2],
  },
  title50: {
    sizes: [fontSizes[9]], // 42
    lineHeights: [1.2],
  },

  // FIXME - THESE STYLES ARE NOW deprecated AND SHOULD NO LONGER BE USED
  title3: {
    // 36
    sizes: [fontSizes[7]],
    // 46
    lineHeights: [lineHeights[5], lineHeights[5], lineHeights[5], lineHeights[5]],
    fontWeight: 'bold',
  },
  title2: {
    // 26
    sizes: [fontSizes[6]],
    // 30
    lineHeights: [lineHeights[4], lineHeights[4], lineHeights[4], lineHeights[4]],
    fontWeight: 'semiBold',
  },
  title1: {
    // 20
    sizes: [fontSizes[5]],
    // 24
    lineHeights: [lineHeights[3], lineHeights[3], lineHeights[3], lineHeights[3]],
    fontWeight: 'semiBold',
  },
  text2: {
    // 16
    sizes: [fontSizes[3]],
    // 20
    lineHeights: [lineHeights[2], lineHeights[2], lineHeights[2], lineHeights[2]],
  },
  text1: {
    // 13
    sizes: [fontSizes[2]],
    // 16
    lineHeights: [lineHeights[1], lineHeights[1], lineHeights[1], lineHeights[1]],
  },
  caption: {
    // 11
    sizes: [fontSizes[1]],
    // 14
    lineHeights: [lineHeights[0], lineHeights[0], lineHeights[0], lineHeights[0]],
  },
}

export const SECONDS_IN_MINUTE = 60
export const SECONDS_IN_HOUR = 3600
export const SECONDS_IN_DAY = 86400
export const MINUTES_IN_HOUR = 60
export const MINUTES_IN_DAY = 1440

export const FORM_DATA_CONTENT_TYPE = 'multipart/form-data'
