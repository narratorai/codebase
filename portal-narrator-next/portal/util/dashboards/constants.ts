export const DASHBOARD_LAYOUT_VERSION_1 = 1
