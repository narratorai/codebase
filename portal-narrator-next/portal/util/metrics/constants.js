// METRIC PAIR IDs - choice for metric pair density
export const METRIC_PAIR_ID_DAY = 'day' // "Today / Yesterday"
export const METRIC_PAIR_ID_WEEK = 'week' // "This Week / Last Week"
export const METRIC_PAIR_ID_MONTH = 'month' // "This Month / Last Month"

// TIME SEGMENT METRIC KEYS - object keys inside metrics.all[METRIC_NAME].data
export const TIME_SEGMENT_METRIC_KEY_OVERALL = 'overall'
export const TIME_SEGMENT_METRIC_KEY_YESTERDAY = 'yesterday'
export const TIME_SEGMENT_METRIC_KEY_DAY = 'day'
export const TIME_SEGMENT_METRIC_KEY_LAST_WEEK = 'last_week'
export const TIME_SEGMENT_METRIC_KEY_WEEK = 'week'
export const TIME_SEGMENT_METRIC_KEY_LAST_MONTH = 'last_month'
export const TIME_SEGMENT_METRIC_KEY_MONTH = 'month'
