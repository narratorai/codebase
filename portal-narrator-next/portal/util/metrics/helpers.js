import _ from 'lodash'
import moment from 'moment'

import {
  METRIC_PAIR_ID_DAY,
  METRIC_PAIR_ID_WEEK,
  METRIC_PAIR_ID_MONTH,
  TIME_SEGMENT_METRIC_KEY_YESTERDAY,
  TIME_SEGMENT_METRIC_KEY_DAY,
  TIME_SEGMENT_METRIC_KEY_LAST_WEEK,
  TIME_SEGMENT_METRIC_KEY_WEEK,
  TIME_SEGMENT_METRIC_KEY_LAST_MONTH,
  TIME_SEGMENT_METRIC_KEY_MONTH,
} from './constants'

// TIME HELPERS
const today = moment().format('dddd, MMMM Do')

const yesterday = moment().subtract(1, 'days').format('dddd, MMMM Do')

const startOfWeek = moment().startOf('week').add(1, 'days').format('MM/DD/YY')

const startOfLastWeek = moment().startOf('week').subtract(1, 'weeks').add(1, 'days').format('MM/DD/YY')

const thisMonth = moment().format('MMMM')

const lastMonth = moment().subtract(1, 'months').format('MMMM')

export const getTimeRange = (metricPairId) => {
  let timeRange

  switch (metricPairId) {
    case METRIC_PAIR_ID_DAY:
      timeRange = `${today} vs. ${yesterday}`
      break
    case METRIC_PAIR_ID_WEEK:
      timeRange = `week of ${startOfWeek} vs. week of ${startOfLastWeek}`
      break
    case METRIC_PAIR_ID_MONTH:
      timeRange = `${thisMonth} vs. ${lastMonth}`
      break
    default:
  }

  return timeRange
}

/*
 * Traverse activty metrics object tree to return metrics object
 * @param metrics - activity.metrics
 * @param metricName - metric name (ex: 'total_events')
 */
export const getMetricObject = (metrics, metricName) => _.get(metrics, `all.${metricName}.data`, {})

/*
 * Get primary segment key based on metricPairId.
 * It's what's appended to the activitiy slug segment url for loaded segment.
 * @param metricPairId - one of "day", "week", "month"
 */
export const getPrimarySegmentKey = (metricPairId) => {
  let key
  switch (metricPairId) {
    case METRIC_PAIR_ID_DAY:
      key = 'today'
      break
    case METRIC_PAIR_ID_WEEK:
      key = 'this_week'
      break
    case METRIC_PAIR_ID_MONTH:
      key = 'this_month'
      break
    default:
  }

  return key
}

/*
 * Get secondary segment key based on metricPairId
 * It's what's appended to the activitiy slug segment url for loaded segment.
 * @param metricPairId - one of "day", "week", "month"
 */
export const getSecondarySegmentKey = (metricPairId) => {
  let key
  switch (metricPairId) {
    case METRIC_PAIR_ID_DAY:
      key = 'yesterday'
      break
    case METRIC_PAIR_ID_WEEK:
      key = 'last_week'
      break
    case METRIC_PAIR_ID_MONTH:
      key = 'last_month'
      break
    default:
  }

  return key
}

/*
 * Get primary label based on metricPairId
 * @param metricPairId - one of "day", "week", "month"
 */
export const getPrimaryLabel = (metricPairId) => {
  let label
  switch (metricPairId) {
    case METRIC_PAIR_ID_DAY:
      label = 'Today'
      break
    case METRIC_PAIR_ID_WEEK:
      label = 'This Week'
      break
    case METRIC_PAIR_ID_MONTH:
      label = 'This Month'
      break
    default:
  }

  return label
}

/*
 * Get secondary label based on metricPairId
 * @param metricPairId - one of "day", "week", "month"
 */
export const getSecondaryLabel = (metricPairId) => {
  let label
  switch (metricPairId) {
    case METRIC_PAIR_ID_DAY:
      label = 'Yesterday'
      break
    case METRIC_PAIR_ID_WEEK:
      label = 'Last Week'
      break
    case METRIC_PAIR_ID_MONTH:
      label = 'Last Month'
      break
    default:
  }

  return label
}

/*
 * Get primary metric value and label from metric object based on metricPairId
 * @param metrics - activity.metrics
 * @param metricName - metric name (ex: 'total_events')
 * @param metricPairId - one of "day", "week", "month"
 */
export const getPrimaryMetric = (metrics, metricName, metricPairId) => {
  const metricObject = getMetricObject(metrics, metricName)

  let value
  switch (metricPairId) {
    case METRIC_PAIR_ID_DAY:
      value = metricObject[TIME_SEGMENT_METRIC_KEY_DAY]
      break
    case METRIC_PAIR_ID_WEEK:
      value = metricObject[TIME_SEGMENT_METRIC_KEY_WEEK]
      break
    case METRIC_PAIR_ID_MONTH:
      value = metricObject[TIME_SEGMENT_METRIC_KEY_MONTH]
      break
    default:
  }

  return value
}

/*
 * Get secondary metric key from the metricPairId
 * @param metricPairId - one of "day", "week", "month"
 */
export const getSecondaryMetricKey = (metricPairId) => {
  let value
  switch (metricPairId) {
    case METRIC_PAIR_ID_DAY:
      value = TIME_SEGMENT_METRIC_KEY_YESTERDAY
      break
    case METRIC_PAIR_ID_WEEK:
      value = TIME_SEGMENT_METRIC_KEY_LAST_WEEK
      break
    case METRIC_PAIR_ID_MONTH:
      value = TIME_SEGMENT_METRIC_KEY_LAST_MONTH
      break
    default:
  }

  return value
}

/*
 * Get secondary metric value and label from metric object based on metricPairId
 * @param metrics - activity.metrics
 * @param metricName - metric name (ex: 'total_events')
 * @param metricPairId - one of "day", "week", "month"
 */
export const getSecondaryMetric = (metrics, metricName, metricPairId) => {
  const metricObject = getMetricObject(metrics, metricName)
  return metricObject[getSecondaryMetricKey(metricPairId)]
}

/*
 * Get both primary and secondary metric values and labels
 * @param metrics - activity.metrics
 * @param metricName - metric name (ex: 'total_events')
 * @param metricPairId - one of "day", "week", "month"
 */
export const getMetrics = (metrics, metricName, metricPairId) => {
  return {
    primaryMetric: getPrimaryMetric(metrics, metricName, metricPairId),
    primaryLabel: getPrimaryLabel(metricPairId),
    secondaryMetric: getSecondaryMetric(metrics, metricName, metricPairId),
    secondaryLabel: getSecondaryLabel(metricPairId),
  }
}
