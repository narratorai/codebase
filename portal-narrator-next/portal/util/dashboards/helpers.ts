import _ from 'lodash'
import { Layout } from 'react-grid-layout'

import { ALL_PLOT_TYPES, CONTENT_TYPE_MARKDOWN } from 'util/narratives/constants'
import { DASHBOARD_LAYOUT_VERSION_1 } from 'util/dashboards/constants'
import {
  DEFAULT_GRID_LAYOUT,
  DEFAULT_GRID_LAYOUT_MIN_WIDTH,
  DEFAULT_GRID_LAYOUT_MIN_HEIGHT,
  DEFAULT_GRID_LAYOUT_PLOT_MIN_WIDTH,
  DEFAULT_GRID_LAYOUT_PLOT_MIN_HEIGHT,
  DEFAULT_GRID_LAYOUT_MARKDOWN_MIN_HEIGHT,
} from 'components/Narratives/Dashboards/BuildDashboard/constants'

interface FormatGridItemDimensionsProps {
  version?: number
  itemLayout?: Layout
  contentType?: string
}

// backfill old dashboard items with a layout that fits
// the grid's dimensions
export const formatGridItemDimensions = ({ version, itemLayout, contentType }: FormatGridItemDimensionsProps) => {
  // if the dashboard items have already been updated
  if (version === DASHBOARD_LAYOUT_VERSION_1) {
    // return the itemLayout as is (backup with Default - but shouldn't happen)
    return itemLayout || DEFAULT_GRID_LAYOUT
  }

  // if dashboard items have never been updated
  if (!_.isFinite(version)) {
    // format item to match version 1 layout

    // plots vs all other content have different minH/minW
    const isPlotContent = _.includes(ALL_PLOT_TYPES, contentType)

    // calculate min height based on content type
    // (plots should be bigger while markdown should be smaller than)
    const minH = getGridItemMinH(contentType)

    return !_.isEmpty(itemLayout)
      ? {
          ...itemLayout,
          h: itemLayout.h * 4 * 0.8,
          y: itemLayout.y * 4 * 0.9,
          w: itemLayout.w * 5 * 0.85,
          x: itemLayout.x * 5 * 0.9,
          minW: isPlotContent ? DEFAULT_GRID_LAYOUT_PLOT_MIN_WIDTH : DEFAULT_GRID_LAYOUT_MIN_WIDTH,
          minH,
        }
      : { ...DEFAULT_GRID_LAYOUT, minH }
  }

  // shouldn't happen, but if there is a bad version or something
  // return either the itemLayout or the default
  return !_.isEmpty(itemLayout) ? itemLayout : DEFAULT_GRID_LAYOUT
}

export const getGridItemMinH = (contentType?: string) => {
  const isPlotContent = _.includes(ALL_PLOT_TYPES, contentType)
  const isMarkdownContent = contentType === CONTENT_TYPE_MARKDOWN

  let minH = DEFAULT_GRID_LAYOUT_MIN_HEIGHT
  if (isPlotContent) {
    minH = DEFAULT_GRID_LAYOUT_PLOT_MIN_HEIGHT
  }

  if (isMarkdownContent) {
    minH = DEFAULT_GRID_LAYOUT_MARKDOWN_MIN_HEIGHT
  }

  return minH
}
