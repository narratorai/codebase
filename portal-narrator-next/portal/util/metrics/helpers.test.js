import { getMetricObject, getPrimaryMetric, getSecondaryMetric } from './helpers'
import { mockActivity } from '../../../test/mockStore'

import { METRIC_PAIR_ID_MONTH } from './constants'

describe('Metrics - helpers', () => {
  describe('#getMetricObject', () => {
    it('gets metric object from metricName', () => {
      expect(getMetricObject(mockActivity.metrics, 'total_events')).toMatchSnapshot()
    })
  })

  describe('#getPrimaryMetric', () => {
    it('gets time density specific metric for current metric', () => {
      expect(getPrimaryMetric(mockActivity.metrics, 'total_events', METRIC_PAIR_ID_MONTH)).toMatchSnapshot()
    })
  })

  describe('#getSecondaryMetric', () => {
    it('gets time density specific metric for last metric', () => {
      expect(getSecondaryMetric(mockActivity.metrics, 'total_events', METRIC_PAIR_ID_MONTH)).toMatchSnapshot()
    })
  })
})
