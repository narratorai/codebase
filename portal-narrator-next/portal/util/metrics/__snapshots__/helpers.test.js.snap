// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics - helpers #getMetricObject gets metric object from metricName 1`] = `
{
  "last_month": 13004,
  "month": 2345,
  "overall": 627521,
}
`;

exports[`Metrics - helpers #getPrimaryMetric gets time density specific metric for current metric 1`] = `2345`;

exports[`Metrics - helpers #getSecondaryMetric gets time density specific metric for last metric 1`] = `13004`;
